internal/nostr/models.go

```diff
@@ -153,13 +153,24 @@ type NIP47Request struct {
 	SignedEvent  *nostr.Event `json:"event"`
 }
 
+type Filter struct {
+	IDs     []string         `json:"ids,omitempty"`
+	Kinds   []int            `json:"kinds,omitempty"`
+	Authors []string         `json:"authors,omitempty"`
+	Tags    nostr.Tags       `json:"tags,omitempty"`
+	Since   *nostr.Timestamp `json:"since,omitempty"`
+	Until   *nostr.Timestamp `json:"until,omitempty"`
```

Comment:
should we use the nostr lib types there or our own types that we then pack into a nostr Filter?